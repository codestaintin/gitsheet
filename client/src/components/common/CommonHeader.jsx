import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Header = ({ sidebarOpen }) => (
  <Fragment>
    <nav
      className="navbar navbar-light"
      style={{ backgroundColor: '#3D4E81', zIndex: sidebarOpen ? 0 : 100 }}
    >
      <ul className="navbar-nav">
        <li className="nav-item">
          <Link className="nav-link link" to="/">Git Sheet</Link>
        </li>
      </ul>

      <ul className="nav justify-content-end">
        <li className="nav-item">
          <Link className="nav-link link" to="/signin">Sign In</Link>
        </li>
        <li className="nav-item">
          <Link className="nav-link link" to="/signup">Sign Up</Link>
        </li>
      </ul>
    </nav>
  </Fragment>
);

Header.propTypes = {
  sidebarOpen: PropTypes.bool
};

export default Header;