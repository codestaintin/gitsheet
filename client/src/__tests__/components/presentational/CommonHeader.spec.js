import React from 'react';
import { shallow } from 'enzyme';
import CommonHeader from '../../../components/common/CommonHeader';

describe('<CommonHeader>', () => {
  const props = {
    sidebarOpen: true
  };
  it('should render CommonHeader component', () => {
    const tree = shallow(<CommonHeader {...props} />);
    expect(tree).toMatchSnapshot();
  });
});