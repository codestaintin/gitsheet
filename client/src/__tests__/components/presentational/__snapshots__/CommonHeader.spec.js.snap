// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommonHeader> should render CommonHeader component 1`] = `
<Fragment>
  <nav
    className="navbar navbar-light"
    style={
      Object {
        "backgroundColor": "#3D4E81",
        "zIndex": 0,
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <Link
          className="nav-link link"
          replace={false}
          to="/"
        >
          Git Sheet
        </Link>
      </li>
    </ul>
    <ul
      className="nav justify-content-end"
    >
      <li
        className="nav-item"
      >
        <Link
          className="nav-link link"
          replace={false}
          to="/signin"
        >
          Sign In
        </Link>
      </li>
      <li
        className="nav-item"
      >
        <Link
          className="nav-link link"
          replace={false}
          to="/signup"
        >
          Sign Up
        </Link>
      </li>
    </ul>
  </nav>
</Fragment>
`;
